proto/internal.proto

```diff
@@ -181,7 +181,60 @@ message InternalLeaderLeaseResponse{
   optional ResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 }
 
+// An InternalRequestUnion contains exactly one of the optional requests.
```

Comment:
Add a comment to the BatchRequest/BatchResponseUnion that whenever something is added there, it needs to be added to the two corresponding internal counterparts.